lib/friendly_id/sequentially_slugged.rb

```diff
@@ -26,7 +26,7 @@ def initialize(scope, slug, slug_column, sequence_separator)
       end
 
       def next_slug
-        slug + sequence_separator + next_sequence_number.to_s
+        [slug, next_sequence_number].compact.join(sequence_separator)
```

Comment:
As for me, it's not good idea to run `next_sequence_number` when slug is `nil`. We can just return `nil` in this case to be sure that slug was not generated at all.